soccer/VisionReceiver.cpp

```diff
@@ -9,15 +9,18 @@
 
 using namespace std;
 
-VisionReceiver::VisionReceiver(bool sim, int port) {
-    simulation = sim;
-    _running = false;
-    this->port = port;
-}
+VisionReceiver::VisionReceiver(bool sim, int port)
+    : simulation(sim), _running(false), port(port) {}
 
 void VisionReceiver::stop() {
-    _running = false;
-    wait();
+    if (isRunning()) {
+        // Handle Case where start has been called, but _running has not been
+        // set to true yet.
+        while (!_running) {
```

Comment:
I'm not 100% sure why this is needed... if isRunning() return's true won't _running also be true? or are they different variables. I really don't like spinlocks :/